src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3544,53 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL)
+                {
+                    uint8_t scanChannel;
+                    uint16_t scanDuration;
+
+                    parsedLength = spinel_datatype_unpack(
```

Comment:
Thanks @darconeous and @abtink for you comments. 

The calling pattern of setting the scan state (which immediately completes) and then getting the result in a separate command/property later works the best for me. I am Ok with using `SPINEL_PROP_MAC_SCAN_PERIOD` for setting the duration of the scan. For the `SPINEL_PROP_MAC_SCAN_MASK` I guess I can just go the route of failing if there is more than one channel bit set.

Is that ok with everyone?